daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Node.scala

```diff
@@ -159,6 +160,9 @@ object Node {
   ) extends Action
       with ActionNodeInfo.Exercise {
 
+    // identified where the choic is defined, in a template if left or in an interface if right
+    def typeId: Either[TypeConName, TypeConName] = interfaceId.toRight(templateId)
```

Comment:
I think using `Either` here is confusing (e.g. why are interfaces `Right` but templates are `Left`? it's arbitrary). I would prefer a custom type with `ByInterface` and `ByTemplate` constructors. Could be done in separate PR though.

Should this be `private[lf]`?

Reply:
Or instead, just drop the `Either` and always return a `TypeConName`. Since you're not actually using the `Either` structure here.

Reply:
I will remove that.